packages/addon/lua/autorun/client/gelly-update-check.lua

```diff
@@ -0,0 +1,73 @@
+include("gelly/ui/markdown-popup.lua")
+
+local GH_RELEASES_API_URL = "https://api.github.com/repos/yogwoggf/gelly/releases/latest"
+local FAVORED_RELEASE_ASSET = "gelly-gmod-release-x64.zip" -- we favor the release build than the debug build
+
+---@alias GellyParsedRelease {version: string, releaseNotes: string, downloadURL: string}
+
+--- Parses the release API response given by GitHub
+---@param responseJSON string The JSON response from the GitHub API
+---@return GellyParsedRelease release The parsed release information
+local function parseReleaseResponse(responseJSON)
+	local parsedTable = util.JSONToTable(responseJSON)
+	if not parsedTable then
+		error("Failed to parse JSON response from GitHub API")
+	end
+
+	local release = {
+		version = parsedTable.tag_name,
+		releaseNotes = parsedTable.body,
+		downloadURL = nil
+	}
+
+	release.releaseNotes = string.gsub(release.releaseNotes, "\r\n", "\n") -- normalize line endings
+
+	for _, assetEntry in ipairs(parsedTable.assets) do
+		if assetEntry.name == FAVORED_RELEASE_ASSET then
+			-- must be the browser_download_url so that we can immediately download it
+			release.downloadURL = assetEntry.browser_download_url
+			break
+		end
+	end
+
+	return release
```

Comment:
No error checking if a release url was found. In fact, this might even be a real-world scenario for a yanked release